check_diff/src/lib.rs

```diff
@@ -56,3 +128,83 @@ pub fn change_directory_to_path(dest: &Path) -> io::Result<()> {
     );
     return Ok(());
 }
+
+// Compiles and produces two rustfmt binaries.
+// One for the current master, and another for the feature branch
+// Parameters:
+// dest: Directory where rustfmt will be cloned
+pub fn compile_rustfmt(
+    dest: &Path,
+    remote_repo_url: String,
+    feature_branch: String,
+    commit_hash: Option<String>,
+) -> Result<Command, CheckDiffError> {
+    const RUSTFMT_REPO: &str = "https://github.com/rust-lang/rustfmt.git";
+
+    let clone_git_result = clone_git_repo(RUSTFMT_REPO, dest);
+    if clone_git_result.is_err() {
+        return Err(CheckDiffError::FailedGit(clone_git_result.err().unwrap()));
+    }
+
+    let git_remote_add_result = git_remote_add(remote_repo_url.as_str());
+    if git_remote_add_result.is_err() {
+        return Err(CheckDiffError::FailedGit(
+            git_remote_add_result.err().unwrap(),
+        ));
+    }
+
+    let git_fetch_result = git_fetch(feature_branch.as_str());
+    if git_fetch_result.is_err() {
+        return Err(CheckDiffError::FailedGit(git_fetch_result.err().unwrap()));
+    }
+
+    let cargo_version = env!("CARGO_PKG_VERSION");
+    info!("Compiling with {}", cargo_version);
```

Comment:
Is this the same as running? This is what the current `check_diff.sh` does.

```
    CARGO_VERSION=$(cargo --version)
    echo -e "\ncompiling with $CARGO_VERSION\n"
```

Reply:
yep! this macro actually gives details about the `cargo` used. `rustc` does not have a similar macro hence a `Command` has to be used. The documentation for `env!` can be found [here](https://doc.rust-lang.org/cargo/reference/environment-variables.html)

Reply:
From the definition given in the docs I don't think `CARGO_PKG_VERSION` is the version of `cargo` that's currently being run. It seems like this would give me the version of the `check_diff` project.

<img width="897" alt="Screen Shot 2024-08-18 at 9 37 54 PM" src="https://github.com/user-attachments/assets/2bab3664-93c5-4482-9cb4-9111f10c5fa7">

Reply:
ah I must have read the docs wrongly. sorry about that.